draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -356,869 +524,832 @@ document are to be interpreted as described in {{RFC2119}}.
+certain algebraic conditions must hold in order to be applied.
 
 ## Notation
 
-As a rough style guide for the following, we use (x, y) to be the output
-coordinates of the encoding method. Indexed values are used when the algorithm
-will choose between candidate values. For example, the SWU algorithm computes
-three candidates (x1, y1), (x2, y2), (x3, y3), from which the final (x, y)
-output is chosen via constant time comparison operations.
+As a rough style guide the following convention is used:
 
-We use u, v to denote the values in Fp output from hash2base, and use as
-initial values in the encoding.
+- All arithmetic operations are performed over the finite field, unless
+  otherwise be explicitly stated, e.g. integer arithmetic.
 
-We use t1, t2, ..., as reusable temporary variables. For notable variables, we
-will use a distinct name, for ease of debugging purposes when correlating with
-test vectors.
+- (x, y): are the coordinates of a point obtained by the encoding method.
+  Indexed values are used when the algorithm calculates some candidate values.
 
-The code presented here corresponds to the example Sage {{SAGE}} code found at
-{{github-repo}}. Which is additionally used to generate intermediate test
-vectors. The Sage code is also checked against the hacspec implementation.
+- u: denotes an element of F produced by the hash2base function and is used
+  as initial value of the encoding.
 
-Note that each encoding requires that certain preconditions must hold in
-order to be applied.
+- t1, t2, ...: are reusable temporary variables. For notable variables,
+    distinct names are used easing the debugging process when correlating with
+    test vectors.
 
-## Encodings for Weierstrass curves
+- c1, c2, ...: are constant values, which can be computed in advance.
 
-The following encodings apply to elliptic curves defined as E: y^2 = x^3+Ax+B,
-where 4A^3+27B^2 != 0.
+## Encodings for Weierstrass curves
 
+The following encodings apply to elliptic curves defined by the equation
+E: y^2 = x^3 + A\*x + B, where 4\*A^3 + 27\*B^2 != 0.
 
 ### Icart Method {#icart}
 
 The map2curve_icart(alpha) implements the Icart encoding method from {{Icart09}}.
 
-**Preconditions**
-
-A Weierstrass curve over F_{p^n}, where p>3 and p^n = 2 mod 3
-(or p = 2 mod 3 and for odd n).
+Preconditions: An elliptic curve over F, such that p>3 and q=p^m=2 (mod 3), or
+p=2 (mod 3) and odd m.
 
-**Examples**
+Input: alpha, an octet string to be hashed.
 
-- P-384
+Constants: A and B, the parameters of the Weierstrass curve.
 
-**Algorithm**: map2curve_icart
-
-Input:
-
- - alpha: an octet string to be hashed.
- - A, B : the constants from the Weierstrass curve.
-
-Output:
-
- - (x,y), a point in E.
+Output: (x, y), a point on E.
 
 Operations:
 
 ~~~
-u = hash2base(alpha)
-v = ((3A - u^4) / 6u)
-x = (v^2 - B - (u^6 / 27))^(1/3) + (u^2 / 3)
-y = ux + v
-Output (x, y)
+1. u = hash2base(alpha)
+2. v = ((3*A - u^4) / 6*u)
```

Comment:
Can we make `3*A` a constant?

Reply:
Done. It was changed in the implementation, but not in the description.